src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -1553,16 +1564,21 @@ internal void OnSequencePointHit(FunctionContext functionContext)
             {
                 if (functionContext._breakPoints[functionContext._currentSequencePointIndex])
                 {
-                    var breakpoints = (from breakpoint in functionContext._boundBreakpoints
-                                       where
-                                           breakpoint.SequencePointIndex == functionContext._currentSequencePointIndex &&
-                                           breakpoint.Enabled
-                                       select breakpoint).ToList<Breakpoint>();
-
-                    breakpoints = TriggerBreakpoints(breakpoints);
-                    if (breakpoints.Count > 0)
+                    if (functionContext._boundBreakpoints.TryGetValue(functionContext._currentSequencePointIndex, out var bps))
```

Comment:
This is the meat of the improvement when looking up BP, instead of looping over all in the file we get it from the map based on sequence point.

Reply:
Do you say it is a single place where we get a performance improvement?
If so I wonder did you try to unroll the Linq?